lib/block-supports/pattern.php

```diff
@@ -13,7 +13,8 @@
 	 * @param WP_Block_Type $block_type Block Type.
 	 */
 	function gutenberg_register_pattern_support( $block_type ) {
-		$pattern_support = 'core/paragraph' === $block_type->name ? true : false;
+		global $block_bindings_allowed_blocks;
+		$pattern_support = array_key_exists( $block_type->name, $block_bindings_allowed_blocks );
```

Comment:
It looks like this global variable will be removed in #57742, so if this PR ships first, it might be nice to help update #57742.

Reply:
Now it fails in trunk
```PHP
Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /plugins/gutenberg/lib/block-supports/pattern.php:17
```

Reply:
Fixed by #57912. I did leave a note on #57742 to say that it needs to be rebased and #57912 merged first, but I don't the person that merged it saw that.

Guess I should've been a bit noisier 😄

Reply:
I just saw it 😅 I didn't realize that https://github.com/WordPress/gutenberg/pull/57742 would break trunk once this one is merged. Thanks for the patch @talldan!

Reply:
No worries @michalczaplinski! Thanks for approving and merging that PR, good to get it across the line!